modules/core/src/main/scala/org/scalasteward/core/gitlab/http4s/Http4sGitlabApiAlg.scala

```diff
@@ -116,15 +141,54 @@ class Http4sGitLabApiAlg[F[_]: MonadThrowable](
 
   def createPullRequest(repo: Repo, data: NewPullRequestData): F[PullRequestOut] = {
     val targetRepo = if (doNotFork) repo else repo.copy(owner = user.login)
-    for {
+    val mergeRequest = for {
       projectId <- client.get[ProjectId](url.repos(repo), modify(repo))
       payload = MergeRequestPayload(url.encodedProjectId(targetRepo), projectId.id, data)
-      res <- client.postWithBody[PullRequestOut, MergeRequestPayload](
+      res <- client.postWithBody[MergeRequestOut, MergeRequestPayload](
         url.mergeRequest(targetRepo),
         payload,
         modify(repo)
       )
     } yield res
+
+    def waitForMergeRequestStatus(internalId: Int, retries: Int = 10): F[MergeRequestOut] =
+      client
+        .get[MergeRequestOut](url.existingMergeRequest(repo, internalId), modify(repo))
+        .flatMap {
+          case mr if (mr.mergeStatus =!= GitlabMergeStatus.Checking) => monad.pure(mr)
+          case _ if (retries > 0)                                    => waitForMergeRequestStatus(internalId, retries - 1)
+          case other                                                 => monad.pure(other)
+        }
+
+    val updatedMergeRequest =
+      if (!mergeWhenPipelineSucceeds)
+        mergeRequest
+      else
+        mergeRequest
+          .flatMap(mr => waitForMergeRequestStatus(mr.iid))
+          .flatMap {
+            case mr if (mr.mergeStatus === GitlabMergeStatus.CanBeMerged) =>
+              for {
+                _ <- logger.info(s"Setting ${mr.webUrl} to merge when pipeline succeeds")
+                res <-
+                  client
+                    .put[MergeRequestOut](
+                      url.mergeWhenPiplineSucceeds(repo, mr.iid),
+                      modify(repo)
+                    )
+                    // it's possible that our status changed from can be merged already,
+                    // so just handle it gracefully and proceed without setting auto merge.
+                    .recoverWith { case UnexpectedResponse(_, _, _, status, _) =>
+                      logger
+                        .warn(s"Unexpected gitlab response setting auto merge: $status")
+                        .flatMap(_ => mergeRequest)
```

Comment:
Flatmapping `mergeRequest` will POST a `MergeRequestPayload` again. Can't we just return the `MergeRequestOut` from `mergeRequest` in line 167? That is:
```suggestion
                        .as(mr)
```

Reply:
Ah yeah, my mistake, thanks!